### ISSUE TITLE
Remark mis-parses nested code blocks in list items

### ISSUE THREAD

**Issue by infotexture** [remarkjs/remark#315]
When code blocks are nested within list items with [CommonMark's](http://spec.commonmark.org/0.28/#indented-code-block) 7-space (3+4=7) indentation rule, `remark` fails to recognize the code blocks and treats them as nested paragraphs in the list item rather than `<code>`.

### Steps to reproduce

This [CommonMark playground example](http://spec.commonmark.org/dingus/?text=1.%20↓%208-space%20indentation%20for%20nested%20%60code%60%20_(8th%20space%20is%20treated%20as%20code)_%0A%0A%20%20%20%20%20%20%20%20code%0A%0A2.%20↓%207-space%20indentation%20for%20nested%20%60code%60%20_(correctly%20aligned)_%0A%0A%20%20%20%20%20%20%20code%0A%0A11.%20↓%208-space%20indentation%20for%202-digit%20list%20item%20_(correctly%20aligned)_%0A%0A%20%20%20%20%20%20%20%20code%0A%0A12.%20↓%207-space%20indentation%20for%202-digit%20list%20item%20_(mis-aligned%2C%20so%20not%20treated%20as%20code)_%0A%0A%20%20%20%20%20%20%20code%0A) shows how CommonMark handles nested code blocks within list items with various indentations.

**TL;DR:** CommonMark recognizes 7-space indents as code blocks in single-digit ordered list items.

### Expected behaviour

> The parsed `remark` AST should be same as the result in the CommonMark playground.

### Actual behaviour

The `remark` parser fails to recognize CommonMark's (3+4=7) indentation for code blocks within list items and treats them as nested paragraphs in the list item rather than `<code>`.

This [Prettier playground example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBGAdAAkMmEmAcAtAM4AOAhmHJgJZQAmCMZM10mAZhAE6ZRxHw6mAAaQGwzAH0AFHhgALTKQpVqRJTC7USJOHQCUkgDpQTmcxfNi4JkwCYsuAOzFylGvUbNWUDt178giLWEjIAtmoE5FxEephk6gBtADSYAMxKbvyGtr6WVhAMuagYOPiuKh4MsN5snDx2BHTUAObUMJgANmod7XBhUtJccJAtUNQAXnEJmNY5pnn5s4U2C6gOZS7K7rTVTCx1-o3NbR3dAjTwA+GR0bFCMynpmSpE82ZL1iYgySAQJAcoERkKAyFwuBAAO4ABTBCGBKDIADcINQ6D8QAkYMh2GROrFfgAjLgUADWcBgAGU3LQWshNABXOC-OgQMA4vEEkC0WJcGDQkktMJkDn45kgABWRAAHgAhElgclUshhOAAGVocFFXIgDJgJD1dm14rucC4yBAwq4pNZkKgGJIWlgAHU0QpkHgAAy-R0QWLOkkkC2O-hmpFa37DACODOowwFZCFIqQuLFv1iEXpXCZ6dpnTgAEUGRB4MbfkxCa66O6kHZyyTqN0oC0AMIQMLCi1QaARkAM2IAFTIhIRqdiAF9x0A) shows how `remark` stumbles on the 7-space indentation.

_(See https://github.com/prettier/prettier/issues/3459 for the initial discussion and additional examples.)_

**Comment by wooorm** [comment]
Thanks for the issue and sorry for the late reply!

That’s definitely a bug. The code [for lists](https://github.com/remarkjs/remark/blob/master/packages/remark-parse/lib/tokenize/list.js) is pretty big and bug prone. It could use a rewrite!

**Comment by MarquiseRosier** [comment]
@wooorm @infotexture any fixes for this issue? I believe this is unresolved: This markdown: 
    ## Then numbered lists:

    1. First install the dependencies:
    \`$ npm install\`
    1. Bar **bold**
    1. Fruits
       1. Apple
       1. Oranges
       1. Lemons
          1. Roman
          1. Greek  
    Muiltiline  
    With \`code\`.
          1. Spanish

Gets rendered as: 

<h2 id="then-numbered-lists">Then numbered lists:</h2>
<ol>
    <li>First install the dependencies: <br><code>$ npm install</code></li> . 
    <li>Bar <strong>bold</strong></li>
    <li>Fruits<ol>
       <li>Apple</li> . 
       <li>Oranges</li>
       <li>Lemons 1. Roman 1. Greek <br>Muiltiline <br>With <code>code</code>. 1. Spanish</li></ol></li></ol>

This fails to recognize the sublist under the Lemons list item.

**Comment by MarquiseRosier** [comment]
Sweet; exciting stuff; I'll keep my eyes peeled! :D

**Comment by MarquiseRosier** [comment]
> Heya, just wanted to give an update about [micromark](https://github.com/micromark/micromark), it’s sort-of a new motor that we’ll soon use in remark to parse markdown. It’s not yet 100% ready but will be relatively soon. The good news is, it fixes this issue! (P.S. see [this twitter](https://twitter.com/wooorm/status/1296480268522577923) thread for some more info!)

That's wonderful news! Thanks a ton and we're super excited. :)

**Comment by MarquiseRosier** [comment]
> Sorry for the wait! I just wanted to share that there’s now a PR that solves this issue: #536.

Woot Woot Woot Woot! Wow that's really impressive; thanks a ton I'll share the news with the team!!!!

**Comment by infotexture** [comment]
Nice to see a solution for this on the horizon. Thanks to @wooorm for following up. 🙏

**Comment by wooorm** [comment]
This is now released in [`remark@13.0.0`](https://github.com/remarkjs/remark/releases/tag/13.0.0)
